Geometry/TrackerNumberingBuilder/plugins/TrackerTopologyEP.cc

```diff
@@ -102,240 +22,99 @@ void
-  ttcphase2.add<unsigned int>("tib_layerStartBit",14);
-  ttcphase2.add<unsigned int>("tib_str_fw_bwStartBit",12);
-  ttcphase2.add<unsigned int>("tib_str_int_extStartBit",10);
-  ttcphase2.add<unsigned int>("tib_strStartBit",4);
-  ttcphase2.add<unsigned int>("tib_moduleStartBit",2);
-  ttcphase2.add<unsigned int>("tib_sterStartBit",0);
-  ttcphase2.add<unsigned int>("tib_layerMask",0x7);
-  ttcphase2.add<unsigned int>("tib_str_fw_bwMask",0x3);
-  ttcphase2.add<unsigned int>("tib_str_int_extMask",0x3);
-  ttcphase2.add<unsigned int>("tib_strMask",0x3F);
-  ttcphase2.add<unsigned int>("tib_moduleMask",0x3);
-  ttcphase2.add<unsigned int>("tib_sterMask",0x3);
-  
-  ttcphase2.add<unsigned int>("tid_sideStartBit",23); 
-  ttcphase2.add<unsigned int>("tid_wheelStartBit",18);
-  ttcphase2.add<unsigned int>("tid_ringStartBit",12);
-  ttcphase2.add<unsigned int>("tid_module_fw_bwStartBit",10);
-  ttcphase2.add<unsigned int>("tid_moduleStartBit",2);
-  ttcphase2.add<unsigned int>("tid_sterStartBit",0);
-  ttcphase2.add<unsigned int>("tid_sideMask",0x3);
-  ttcphase2.add<unsigned int>("tid_wheelMask",0xF);
-  ttcphase2.add<unsigned int>("tid_ringMask",0x3F);
-  ttcphase2.add<unsigned int>("tid_module_fw_bwMask",0x3);
-  ttcphase2.add<unsigned int>("tid_moduleMask",0xFF);
-  ttcphase2.add<unsigned int>("tid_sterMask",0x3);
-  
-  ttcphase2.add<unsigned int>("tob_layerStartBit",20);
-  ttcphase2.add<unsigned int>("tob_rod_fw_bwStartBit",12);
-  ttcphase2.add<unsigned int>("tob_rodStartBit",12);  
-  ttcphase2.add<unsigned int>("tob_moduleStartBit",2);
-  ttcphase2.add<unsigned int>("tob_sterStartBit",0);
-  ttcphase2.add<unsigned int>("tob_layerMask",0xF);
-  ttcphase2.add<unsigned int>("tob_rod_fw_bwMask",0x0);
-  ttcphase2.add<unsigned int>("tob_rodMask",0xFF);
-  ttcphase2.add<unsigned int>("tob_moduleMask",0x3FF);
-  ttcphase2.add<unsigned int>("tob_sterMask",0x3);
-
-  descriptions.add( "trackerTopology2023Constants", ttcphase2 );
+  descriptions.add( "trackerTopology", ttc );
 }
 
-//
-// member functions
-//
-
-// ------------ method called to produce the data  ------------
 TrackerTopologyEP::ReturnType
-TrackerTopologyEP::produce(const IdealGeometryRecord& iRecord)
+TrackerTopologyEP::produce( const TrackerTopologyRcd& iRecord )
 {
-  edm::LogInfo("TrackerTopologyEP") <<  "TrackerTopologyIdealEP::produce(const IdealGeometryRecord& iRecord)";
-    
-  ReturnType myTopo(new TrackerTopology(pxbVals_, pxfVals_, tecVals_, tibVals_, tidVals_, tobVals_));
+  edm::LogInfo("TrackerTopologyEP") <<  "TrackerTopologyEP::produce(const TrackerTopologyRcd& iRecord)";
+  edm::ESHandle<PTrackerParameters> ptp;
+  iRecord.getRecord<PTrackerParametersRcd>().get( ptp );
+  fillParameters( *ptp );
+  
+  ReturnType myTopo( new TrackerTopology( pxbVals_, pxfVals_, tecVals_, tibVals_, tidVals_, tobVals_ ));
 
   return myTopo ;
 }
 
+void
+TrackerTopologyEP::fillParameters( const PTrackerParameters& ptp )
+{
+  pxbVals_.layerStartBit_ = ptp.vitems[0].vpars[0]; // 16
```

Comment:
Thanks, @slava77 - yes, we've discussed and agreed at a tracker DPG meeting (sometime in November-December) that code improvements including this schema incapsulation will come after their feedback. I hope, there will be more comments.